### PR TITLE
Ignore local environment .tool-versions configuration file

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related PRs

- https://github.com/WordPress/gutenberg/pull/38013

## Description 

The `.tool-versions` file configures one's local environment and should likely not be committed. Its format is defined by [asdf-vm](https://asdf-vm.com), and is similar to [nvm](https://github.com/nvm-sh/nvm), [rvm](https://rvm.io), or [rbenv](https://github.com/rbenv/rbenv). Ignoring this file will help ensure the configuration is not accidentally committed to the repository.

To test: n/a

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Comment by hypest** [comment]
👋 @dcalhoun ! I'm not familiar with asdf-vm and .tool-version in particular and I wonder, can you share some quick usage of it? Would that also help with testing/verifying this PR?

**Comment by dcalhoun** [comment]
> 👋 @dcalhoun ! I'm not familiar with asdf-vm and .tool-version in particular and I wonder, can you share some quick usage of it? Would that also help with testing/verifying this PR?

@hypest [`asdf-vm`](https://asdf-vm.com/guide/introduction.html#how-it-works) is an alternative to version managers like [`nvm`](https://github.com/nvm-sh/nvm), [`n`](https://github.com/tj/n), [`rbenv`](https://github.com/rbenv/rbenv), or [`rvm`](https://rvm.io). `asdf-vm` seeks to provide a single tool for managing versions of various runtimes, rather than needing individual tools for each runtime. In the context of this project, `.tool-versions` serves as an alternative to `.nvmrc`, but this PR _does not attempt to replace_ the current usage of `.nvmrc`. 

```bash
❯ echo "14.17.6" > .nvmrc
❯ nvm use
❯ node --version
v14.17.6

❯ echo "nodejs 14.17.6" > .tool-versions
❯ asdf current nodejs
nodejs          14.17.6         /Users/[REDACTED]/Sites/a8c/gutenberg/.tool-versions
❯ node --version
v14.17.6
```

I do not believe there are actionable testing steps for this PR, as it is not proposing any configuration change that impacts the project. The proposed changes relate to Git configuration and help avoid accidental erroneous commits of a tool that I happen to use on my development machine. I would compare the subject to the act of [ignoring VSCode editor configuration files](https://github.com/wordpress-mobile/gutenberg-mobile/blob/628e86fb4cb3ec3d15d0dc2edd75ce834b2fca0a/.gitignore#L100-L105).

**Comment by hypest** [comment]
Gotcha, thanks for elaborating David!

I'll go ahead and LGTM the PRs.

File: .gitignore [hunk 1]
```diff
@@ -124,4 +125,4 @@ ios/vendor
 i18n-test/
 
 # i18n cache
-src/i18n-cache/
\ No newline at end of file
+src/i18n-cache/
```
Comment (hypest) [inline .gitignore:128]: Nit and not a blocker for LGTM'ing the PR: what do you think about undoing the diff of the file ending, or separating it into a commit of its own to help with code archeology? :)

**Review by hypest — APPROVED** [review]
LGTM David, only left a [non blocking comment](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4476/files#r786214151). Thanks!

**Pull request merged**